src/etc/inc/plugins.inc.d/dhcpd.inc

```diff
@@ -1822,8 +1822,8 @@ function dhcpd_dhcrelay6_configure($verbose = false)
         }
 
         if (empty($destif)) {
-            foreach ($a_gateways as $gateway) {
-                if (isset($gateway['defaultgw']) && $gateway['ipprotocol'] == 'inet6') {
+            foreach ((new \OPNsense\Gateways\Gateway())->gatewayIterator() as $gateway) {
```

Comment:
a_gateways is already using `gatewaysIndexedByName()`. Not sure if a new class and different approach makes sense here

Reply:
FWIW, this will also iterate over disabled gateways, which the former code did not (I think that should be the way).